### DISCUSSION TITLE
Can the same be used for managed C++? If not create it?

### DISCUSSION THREAD

**Discussion by OhSoGood** [devlooped/GitInfo#172] [Ideas]
Excellent work, kzu. Nice and simple, versus other complex and black-box like solutions. Brilliant (like a Moq :) )...

Can you use this nuget package for managed c++? If not could you extend it ?

Regards

**Comment by kzu** [comment]
You'd have to create the equivalent of https://github.com/kzu/GitInfo/blob/master/src/GitInfo/build/GitInfo.cs.pp for the C++ you want to generate and send a PR.

You can use the MSBuild properties in any way you can in the C++ project file, though.

**Comment by TravisDean** [comment]
I actually haven't managed to get this to work in a C++ project file.

For some reason, even if you set the variable GitThisAssembly to true, the provided variables are never populated, report is never printed (with importance set to high).

Relevant log lines:
```

Task "Error" skipped, due to false condition; (!Exists('..\packages\GitInfo.2.0.11\build\GitInfo.targets')) was evaluated as (!Exists('..\packages\GitInfo.2.0.11\build\GitInfo.targets')).

Property reassignment: $(GitInfoReportImportance)="high" (previous value: "low") at [....]

Building with tools version "15.0".
Target "SetGitExe" in file "[...]..\packages\GitInfo.2.0.11\build\GitInfo.targets" from project "[....].vcxproj" (entry point):
Task "Exec"
  git version 2.17.1.windows.1
Done executing task "Exec".
Task "Exec" skipped, due to false condition; ('$(GitExe)' == '') was evaluated as ('git' == '').
Done building target "SetGitExe" in project "RetiVueWF.vcxproj".

```
Then no other reference to GitInfo.

I've tried adding GitInfo as a dependency for my main Build target, but no luck.

I think that different properties are assigned and used for building in VC++, as CoreCompileDependsOn is never reassigned to have the GitInfo targets injected.

**Comment by kzu** [comment]
Please use `/bl` when building and the most excelent `http://msbuildlog.com` tool to learn how the VC++ targets are building and to determine what the best hook would be for that scenario. Then add a target that runs before one of the main VC++ main targets before compilation happens, and add a target to the `.vcxproj` that runs before than and depends on `GitVersion`. That should trigger the GitInfo targets. 

Once that works and you have the equivalent of the `.cs.pp`, please send a PR 👍 

Thanks!

**Comment by kzu** [comment]
hey @OhSoGood @TravisDean any chance you could give this PR a try for this scenario? https://github.com/kzu/GitInfo/pull/130

🙏 

thanks!
